interpreter/interpreter.go

```diff
@@ -136,7 +136,36 @@ func (itrp *Interpreter) doPackageClause(pkg *semantic.PackageClause) error {
 	return nil
 }
 
-func (itrp *Interpreter) doImport(dec *semantic.ImportDeclaration, scope values.Scope, importer Importer) error {
+type key int
+
+const packagesKey key = iota
+
+type Packages map[string]*Package
+
+func (p Packages) Inject(ctx context.Context) context.Context {
+	return context.WithValue(ctx, packagesKey, p)
+}
+
+func GetPackages(ctx context.Context) Packages {
+	v := ctx.Value(packagesKey)
+	if v == nil {
+		return nil
+	}
+	return v.(Packages)
+}
+
+func GetOption(ctx context.Context, pkg string, option string) (values.Value, bool) {
+	packages := GetPackages(ctx)
+	if packages == nil {
+		return values.InvalidValue, false
+	}
+	if pkg, ok := packages[pkg]; ok {
+		return pkg.object.Get(option)
+	}
+	return values.InvalidValue, false
+}
+
+func (itrp *Interpreter) doImport(dec *semantic.ImportDeclaration, ctx context.Context, scope values.Scope, importer Importer) error {
```

Comment:
_nit_ The context is typically the first parameter, as mentioned on Go's stdlib docs:
https://pkg.go.dev/context@go1.17.3#pkg-overview

(We seem to break the rules outlined there in several places though)